src/pages/ThorChainLP/components/ReusableLpStatus/ReusableLpStatus.tsx

```diff
@@ -113,10 +113,11 @@ export const ReusableLpStatus: React.FC<ReusableLpStatusProps> = ({
   // This allows us to either do a single step or multiple steps
   // Once a step is complete the next step is shown
   // If the active step is the same as the length of steps we can assume it is complete.
-  const isComplete = useMemo(
-    () => activeStepIndex === txAssets.length,
-    [activeStepIndex, txAssets.length],
-  )
+  const isComplete = activeStepIndex === txAssets.length
+
+  const isFailed = txStatus === TxStatus.Failed
+
+  const isSubmitted = txStatus === TxStatus.Pending
```

Comment:
memoize?

Reply:
Not needed, memoizing primitives values is not needed as it won't cause any unwanted rerenders, you can verify it using [this playground](https://reactplayground.vercel.app/#N4IgLgziBcBmCGAbCBTANCAbrK1QEsA7AExQA8A6AK1xHwFsAHAewCcwACAQUcY9lbN6HAOQUA9D0bUIZEQB1CDFuw4AlFPADGnAUNGtNOhUqZtOwDlsPwwKNc2acAvv0HCRNnQFpiQ8VqI+CiEYCaKWsyEEJyRofBEKKwcALwcfloArvQhYBQA5ihgAKKIKDmhAEIAngCSxAAUno5hAJQRUTEcgk6pVjZ2Dk4NcWAJhEnthD15hiRJDYoc6kZ51pp2peW5i4TLyxraeQDKYKz4OgCyzKRoS-uEmYiId3v7hzoU67YoWxVgDSkU2WU1aIAwUhkZBgdDMqg+YDQHEsmVQpx+SNRKEu5WYrj0Hi8YUUynMHBi1TKxDc+hEFKp3kiZgmoQg4UIcS6Uj6DVaqQAfMj7pzOABtSKZUJI1BgADCzElYAAun0sei7A0AAzAqydMV2MiI8lFAAq5GVqrRYw1IhEOpFHFF8FYrHg1WlRS4LrdKrSautKAaoqV9r1jpiPwA8gAjKgoHQesAxuM6X0cf0-BrAOwxaCiZgAaxEzimimFYfwEEjEz6EtCqRSaQAjOXopxK2aumkDZxG2kRDniW8HR2AO7MAAyIXyYAAFn1na7qhQyoQZ-O+xwAEytrqVyMFvoRuzJ+N5QcN-uF9nLLE4+jMBq8gVCt7LTnMMor5j5JpaWfwGuKDECISL7hMOrOEiorgSgIZlm+6aoPej7PikgrAPc76dJ+KDfr+Ij-oBhTUj2ohgRAnZgJB0EdigMTwYQ9xYUh2K4k+fLoa++y6tEuH4X+AFAdSi5uhwq7ruRHBjpO05zjqyxQY6MlTmuc6McxiF3uxaEYSxH5fogP5NMwsZnlYQkkVJ+4FgpHBKTBVYFoxyyacshhgJkrB7LsAA8-Isb5lSZGAYBRBwUSykEWgFikwC6cacoKqEIzJZwADUHBNq0zgBYhyy+RAjCAd0uEpPIsL5BVHDOvg8DeIg8DRigiDlSARLeLiVD4BV-KANlkgDwf754hFYBGF1mAziBeIwWhVEeWufl+xBSFYV7JF0WxfFnGCjKZqGg0ZGZQO9FtLlLEFaNeyCGUbUMFVIA1ec9WNc1rUVR1XU9SA-VDSNxWEIKsqWSgHBhaIF6ZYOJg8cNs1rYKPH7PcK1zethBRRcW0JTKXpLkGFCE6J7qiIBTizkkHAtdsoQiCG51LYVAOlbdFX3dVtUvU1LVtZ9D7db1g3DVdgpcMQ1LU-8YPME9S7TfD80IbDCvo5jMVxTjRSnjoWaDnmIiEL014lgzSMcEzJU3Sgd30A9T11Q13Pve1qydfz32-cLANAyDEVmTo3QoLASQhFoKDy6tiuEMNC0cFMU1MRyYYq0eYCUsBFDRpHhAAAb3LAURgN4ED4AAXigeZboY9AANyKHnifkConCkAgTycFIIDOM4QA)

Reply:
Ok cool, I always get confused. We have had the discussion of just "memoize all the things" to not have to think about it or worry since the memoization overhead is pretty negligible just for historical context. Might be a good one to bring up in our next sync just for posterity

Reply:
> Ok cool, I always get confused. We have had the discussion of just "memoize all the things" to not have to think about it or worry since the memoization overhead is pretty negligible just for historical context. Might be a good one to bring up in our next sync just for posterity

I seen a commit from @0xApotheosis recently that was doing the exact same thing as I did in this PR, this is kinda why I decided to follow the same rule, It might be worth to talk about it and write the decision somewhere as you said 👍 

I would follow the "Memo everything except primitive values compares" rule on my side 👀

Reply:
Yeah memo isn't so much about the expression itself being referentially stable (which primitives, i.e a boolean here, are), but in the case of memos, ensuring their inner logic doesn't run too much either. Since we're talking about an eqeqeq which is so cheap it's pretty much free, I have nothing against that, no need to `useMemo()` on this guy IMO!